core/src/main/scala/filodb.core/store/ChunkSource.scala

```diff
@@ -198,7 +198,26 @@ trait ChunkSource extends RawChunkSource with StrictLogging {
     }
   }
 
-  def scanTsCardinalities(ref: DatasetRef, shard: Seq[Int],
+  val FILODB_PARTITION_KEY = "filodb.partition"
+
+  /**
+   * Additional check to ensure the query is for the given filodb partition
+   *
+   * @param queryContext QueryContext containing traceInfo of the originating partition
+   * @return
+   */
+  def isCorrectPartitionForCardinalityQuery(queryContext: QueryContext, filodbPartition: String): Boolean = {
+    if (
+      queryContext.traceInfo.contains(FILODB_PARTITION_KEY) &&
```

Comment:
So the only invalid case is that we find a `trace info` does not match the `filodbPartition` from the `partition key`?

Reply:
Yes, that is the only invalid case. This is done to make sure the existing api flows in which TsCardinalities is queried is not impacted.